pkg/mutation/path/parser/node.go

```diff
@@ -112,3 +138,24 @@ func (l List) Value() (string, bool) {
 	}
 	return *l.KeyValue, true
 }
+
+func (l List) String() string {
+	key := quote(l.KeyField)
+	if l.Glob {
+		return fmt.Sprintf("[%s: *]", key)
+	}
+	if l.KeyValue != nil {
+		value := quote(*l.KeyValue)
+		return fmt.Sprintf("[%s: %s]", key, value)
+	}
+	// Represents an improperly specified List node.
+	return fmt.Sprintf("[%q: ]", key)
+}
+
+// quote adds double quotes around the passed string.
+func quote(s string) string {
```

Comment:
I think this function might be too simplistic. I added the following use cases to `Test_Parser` and they both fail:

```
{
	input: `"token-with-trailing-backslash\\"`,
},
{
	input: `"token-with-\\embedded-backslash"`,
},
```

Reply:
Fixed.

Reply:
All I had to do was also escape backslashes so that the parser would properly parse backslashes.